src/vs/workbench/contrib/terminalContrib/chat/browser/terminalChatActions.ts

```diff
@@ -0,0 +1,330 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Codicon } from 'vs/base/common/codicons';
+import { KeyCode, KeyMod } from 'vs/base/common/keyCodes';
+import { localize2 } from 'vs/nls';
+import { ContextKeyExpr } from 'vs/platform/contextkey/common/contextkey';
+import { KeybindingWeight } from 'vs/platform/keybinding/common/keybindingsRegistry';
+import { TerminalSettingId } from 'vs/platform/terminal/common/terminal';
+import { AbstractInlineChatAction } from 'vs/workbench/contrib/inlineChat/browser/inlineChatActions';
+import { CTX_INLINE_CHAT_EMPTY, CTX_INLINE_CHAT_FOCUSED, CTX_INLINE_CHAT_HAS_PROVIDER } from 'vs/workbench/contrib/inlineChat/common/inlineChat';
+import { isDetachedTerminalInstance } from 'vs/workbench/contrib/terminal/browser/terminal';
+import { registerActiveXtermAction } from 'vs/workbench/contrib/terminal/browser/terminalActions';
+import { TerminalContextKeys } from 'vs/workbench/contrib/terminal/common/terminalContextKey';
+import { MENU_TERMINAL_CHAT_INPUT, MENU_TERMINAL_CHAT_WIDGET, MENU_TERMINAL_CHAT_WIDGET_FEEDBACK, MENU_TERMINAL_CHAT_WIDGET_STATUS, TerminalChatCommandId, TerminalChatContextKeys, TerminalChatResponseTypes } from 'vs/workbench/contrib/terminalContrib/chat/browser/terminalChat';
+import { TerminalChatController } from 'vs/workbench/contrib/terminalContrib/chat/browser/terminalChatController';
+
+registerActiveXtermAction({
+	id: TerminalChatCommandId.Start,
+	title: localize2('startChat', 'Start in Terminal'),
+	keybinding: {
+		primary: KeyMod.CtrlCmd | KeyCode.KeyI,
+		when: ContextKeyExpr.and(TerminalContextKeys.focusInAny),
+		weight: KeybindingWeight.WorkbenchContrib,
+	},
+	f1: true,
+	category: AbstractInlineChatAction.category,
+	precondition: ContextKeyExpr.and(
+		ContextKeyExpr.has(`config.${TerminalSettingId.ExperimentalInlineChat}`),
+		ContextKeyExpr.and(TerminalContextKeys.processSupported, TerminalContextKeys.focusInAny),
+		CTX_INLINE_CHAT_HAS_PROVIDER
```

Comment:
fyi - this is the wrong context key. It checks for inline chat providers, not for agents/participants